### ISSUE TITLE
Subtraction of days is inconsistent across DST boundaries

### ISSUE THREAD

**Issue by stieg** [moment/moment#5625]
**Describe the bug**
Moment is not accounting for the hour shift when doing subtract of days across a timezone boundary.  Below is an example that can be seen when running in the 'America/Denver' timezone. 

**To Reproduce**
```js
var moment = require('moment');
var d1 = new Date('2020-04-01');
var d2 = moment(d1).subtract(31, 'days').toDate();
console.log('D2:', d2);
```

**Desktop (please complete the following information):**
 - OS: Fedora 31 (x86_64)
-  Node: v12.6.3
 - Moment: 2.25.3

**Moment-specific environment**
* The time zone setting of the machine the code is running on
* The time and date at which the code was run
* Other libraries in use (TypeScript, Immutable.js, etc)

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
console.log((new Date()).toLocaleString())
console.log((new Date()).getTimezoneOffset())
console.log(moment.version)
```
```
Fri Jun 26 2020 15:10:10 GMT-0600 (Mountain Daylight Time)
6/26/2020, 3:10:10 PM
360
2.25.3
```

**Comment by MinusFour** [comment]
Looking at the `moment-timezone` tests, it looks like that's the intended behavior (time remaining the same).

https://github.com/moment/moment-timezone/blob/develop/tests/moment-timezone/manipulate.js#L21

**Comment by stieg** [comment]
@MinusFour I see.  So here are two examples with their time zones listed:

### Pacific/Honolulu
```sh
➜  af-ohare git:(password-reset-update) ✗ node packages/af-services/test.js
Sat Jun 27 2020 06:13:27 GMT-1000 (Hawaii-Aleutian Standard Time)
Moment Version: 2.25.3
D2: 2020-03-01T00:00:00.000Z
```

### America/Denver
```sh
➜  af-ohare git:(password-reset-update) ✗ node packages/af-services/test.js
Sat Jun 27 2020 10:16:18 GMT-0600 (Mountain Daylight Time)
Moment Version: 2.25.3
D2: 2020-03-01T01:00:00.000Z
```

Notice how the value of D2 in the Denver time zone is at 01:00, but for Pacific/Honolulu the value is 00:00.  This is because America/Denver has a time-shift forward that happened on March 8.  Pacific/Honolulu does not have such a shift.

**Comment by MinusFour** [comment]
Yes, that makes sense. 00:00 UTC is 18:00 MDT (UTC -6). `moment-timezone` is still returning 18:00 but now in MST (which is UTC -7). Therefore 18:00 MST is 01:00 UTC. `18:00 MDT -> 18:00 MST | 00:00 UTC -> 01:00 UTC`

**Comment by stieg** [comment]
Perhaps the definition of a day needs to be clarified here.  When we say we are subtracting one day, does this mean that we are subtracting a days worth of time (24 hours) or are we going back to the exact same hour, minute, second, millisecond yesterday?  Because when it comes to this scenario the two are clearly not the same.  I personally would expect the latter with moment, especially because I am staying within the same timezone.  If this expectation is not correct then documentation needs to be updated to clarify this distinction.  Otherwise if the latter case is what is desired then there is a logic issue here.

**Comment by MinusFour** [comment]
It's actually not even something specific to `moment-timezone` but `moment` itself. Have a look at the documentation for [`add`](https://momentjs.com/docs/#/manipulating/add/).

> There are also special considerations to keep in mind when adding time that crosses over daylight saving time. If you are adding years, months, weeks, or days, the original hour will always match the added hour.

If you want to be more specific though you can use hours instead of days.

**Comment by mattjohnsonpint** [comment]
Days are calendar units, so subtracting them preserves local time while the UTC offset may change across DST. I'd subtract `31 * 24` hours for exactly 744 elapsed hours, or 31 days for the same local clock time 31 calendar days earlier. I'm closing this as expected behavior.
